tools/applink/src/devConsole/action.ts

```diff
@@ -0,0 +1,48 @@
+// (C) 2020 GoodData Corporation
+import { getSourceDescriptor } from "../base/sourceDiscovery";
+import { getTargetDescriptor } from "../base/targetDiscovery";
+import { TerminalUi } from "./ui/ui";
+import { GlobalEventBus, sourceInitialized, targetSelected } from "./events";
+import { ChangeDetector } from "./pipeline/changeDetector";
+import { BuildScheduler } from "./pipeline/buildScheduler";
+import { PackageBuilder } from "./pipeline/packageBuilder";
+import { PackagePublisher } from "./pipeline/publisher";
+
+export async function devConsole(targetDir: string): Promise<number> {
+    const sourceDescriptor = await getSourceDescriptor(
+        (pkg) => !pkg.projectFolder.startsWith("examples") && !pkg.projectFolder.startsWith("skel"),
+    );
+
+    if (!sourceDescriptor) {
+        return 1;
+    }
+
+    const targetDescriptor = getTargetDescriptor(targetDir, sourceDescriptor);
+
+    if (!targetDescriptor.dependencies.length) {
+        console.info("The target project does not have any dependencies on the SDK. There is nothing to do.");
+
+        return 1;
+    }
+
+    /*
+     * Initialize the terminal UI - this will make the app run forever until user triggers exit
+     */
+    new TerminalUi();
```

Comment:
Esoteric: I'm not a fan of constructors with side-effects, but I understand it works here. How about static method?

```ts
TerminalUi.init();
```

Reply:
fixed